flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/rules/datastream/DataStreamAggregateRule.scala

```diff
@@ -159,9 +152,25 @@ class LogicalWindowAggregateRule
     }
     false
   }
+
+  private def rewriteTimeIndicatorOperators(agg: LogicalAggregate, groupExprIdx: Int) = {
+    val project = agg.getInput.asInstanceOf[HepRelVertex].getCurrentRel.asInstanceOf[LogicalProject]
+    val newProjectExpr = mutable.ArrayBuffer[RexNode]()
+    newProjectExpr.appendAll(project.getChildExps)
+    val rexBuilder = agg.getCluster.getRexBuilder
+    newProjectExpr(groupExprIdx) = rexBuilder.makeTimestampLiteral(
+      DataStreamAggregateRule.TIMESTAMP_ZERO, 3)
+    val newProject = project.copy(project.getTraitSet, project.getInput,
+      newProjectExpr, project.getRowType)
+
+    agg.copy(agg.getTraitSet, List(newProject)).asInstanceOf[LogicalAggregate]
```

Comment:
Can we create here a `LogicalAggregate` with adapted `groupSet`? 
I think adding the `windowingGroupSet` to the `LogicalWindowAggregate` is not a very nice solution. It would be better if we could keep the existing code as it is without introducing workarounds for the SQL case.